librad/src/net/gossip.rs

```diff
@@ -64,16 +66,19 @@ pub use storage::*;
 pub use types::*;
 
 #[derive(Clone)]
-pub enum ProtocolEvent {
-    SendAdhoc(Box<Hello>),
-    Connect(Box<Hello>),
+pub enum ProtocolEvent<A> {
+    SendAdhoc(Box<Hello<A>>),
+    Connect(Box<Hello<A>>),
     Disconnect(PeerId),
 }
 
+// Nb.: this type exists only because clippy otherwise complains that
+// `ProtocolEvent`'s size might be sub-optimal (enum size is determined by the
+// largest variant).
```

Comment:
I think that lint isn't so useful here: it's just saying that there's high variance in the size of your enum fields, but that's ok imo. With the `Box` you're creating an indirection which shouldn't be necessary unless you have a very specific performance requirement around this enum, or you want to optimize for the `Disconnect` case here.

Reply:
Yeah... I wonder _when_ this actually matters, as the lint is enabled by default. I would rather disable it for the entire crate if this is something one should only pay attention to in very specific cases, instead of annotating the exceptions everywhere.

Reply:
Ah so this is part of the "default" lints? Sounds "pedantic" to me :thinking: 

I'm thinking: if you need to pack this enum in contiguous memory, eg. a `Vec`, then you don't want to waste space if only one of your variants takes lots of space, and all the others are small. In that case it would make sense to have an indirection for the "large" variant, especially if you don't access it as often as the "small" variants. Eg:

```rust
enum Command {
   OpenConnection(Box<HugeStruct>), // only used once per connection
   Send(SmallStruct), // used a lot
}
struct CommandQueue(Vec<Command>);
```

Maybe in the above case it makes sense to box the `HugeStruct`, if you have millions of these, as you'll rarely pattern-match against `OpenConnection`?

Reply:
Hm yeah. It's the other way round in this case: the small variant should be used less. 

Anyways, I reverted this .. and guess what? `clippy` decided to no longer complain about it :dumbfounded:

Reply:
Yeah that's what I thought, so really don't think it makes sense.